plutus-ledger-api/src/PlutusLedgerApi/V3/Contexts.hs

```diff
@@ -0,0 +1,577 @@
+{-# LANGUAGE DeriveAnyClass    #-}
+{-# LANGUAGE DerivingVia       #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards   #-}
+{-# LANGUAGE TemplateHaskell   #-}
+{-# OPTIONS_GHC -Wno-simplifiable-class-constraints #-}
+{-# OPTIONS_GHC -fno-omit-interface-pragmas #-}
+{-# OPTIONS_GHC -fno-specialise #-}
+{-# OPTIONS_GHC -fno-strictness #-}
+
+module PlutusLedgerApi.V3.Contexts where
+
+import GHC.Generics (Generic)
+import Prettyprinter (nest, vsep, (<+>))
+import Prettyprinter.Extras
+
+import PlutusLedgerApi.V2 qualified as V2
+import PlutusTx qualified
+import PlutusTx.AssocMap hiding (filter, mapMaybe)
+import PlutusTx.Prelude qualified as PlutusTx
+
+import Prelude qualified as Haskell
+
+newtype ColdCommitteeHash = ColdCommitteeHash V2.Credential
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+  deriving (Pretty) via (PrettyShow ColdCommitteeHash)
+  deriving newtype
+    ( PlutusTx.Eq
+    , PlutusTx.ToData
+    , PlutusTx.FromData
+    , PlutusTx.UnsafeFromData
+    )
+
+newtype HotCommitteeHash = HotCommitteeHash V2.Credential
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+  deriving (Pretty) via (PrettyShow HotCommitteeHash)
+  deriving newtype
+    ( PlutusTx.Eq
+    , PlutusTx.ToData
+    , PlutusTx.FromData
+    , PlutusTx.UnsafeFromData
+    )
+
+newtype DRepHash = DRepHash V2.Credential
```

Comment:
My bad, I think I should have named it credential, rather than Hash. I have no idea how, but I mistakenly thought that's how in Plutus calls Credentials :facepalm: 
```suggestion
newtype DRepCredential = DRepCredential V2.Credential
```